src/sinks/splunk_hec/logs/sink.rs

```diff
@@ -193,7 +193,7 @@ impl Partitioner for EventPartitioner {
         let host = self
             .host_key
             .as_ref()
-            .and_then(|host_key| item.event.get(host_key.as_str()))
+            .and_then(|host_key| item.event.get((PathPrefix::Event, host_key)))
```

Comment:
Just noting here that this is not going to work for LogNamespacing. With an event in the Vector namespace, we should really get the host based on the field with the host meaning rather than a configured host name.

This isn't a regression, so doesn't need changing here. I'll raise a separate issue so we can fix this at another point.

Reply:
https://github.com/vectordotdev/vector/issues/18010

Reply:
Good catch.